app/src/main/java/com/github/freeman/bootcamp/DrawHatActivity.kt

```diff
@@ -0,0 +1,361 @@
+                            it.asAndroidBitmap().compress(Bitmap.CompressFormat.PNG, 100, byteArrayOutputStream)
+                            val byteArray = byteArrayOutputStream.toByteArray()
+                            val tempFile = File.createTempFile("image", ".png")
+                            tempFile.writeBytes(byteArray)
+                            hatRef.putFile(tempFile.toUri())
+                            hat.value = it.asAndroidBitmap()
+                        }
+                    }
+                ) { undoCount, redoCount ->
+                    colorBarVisibility.value = false
+                    undoVisibility.value = undoCount != 0
+                    redoVisibility.value = redoCount != 0
+                }
+            }
+        }
+    }
+}
+
+
+// Note: Please see the comments for DrawHatScreen
+@Composable
+private fun DrawHatControlsBar(
+    drawController: DrawController,
+    onColorClick: () -> Unit,
+    onWidthClick: () -> Unit,
+    undoVisibility: MutableState<Boolean>,
+    redoVisibility: MutableState<Boolean>,
+    colorValue: MutableState<Color>,
+    onEraseClick: () -> Unit,
+    currentColor: MutableState<Color>,
+    currentWidth: MutableState<Float>,
+) {
+    val context = LocalContext.current
+    val isToggled = remember { mutableStateOf(false) }
+
+    Row(modifier = Modifier.padding(12.dp), horizontalArrangement = Arrangement.SpaceAround) {
+        MenuItems(
+            R.drawable.ic_undo,
+            LocalContext.current.getString(R.string.undo),
+            if (undoVisibility.value) MaterialTheme.colors.primary else MaterialTheme.colors.secondary
+        ) {
+            if (undoVisibility.value) drawController.unDo()
+        }
+        MenuItems(
+            R.drawable.ic_redo,
+            LocalContext.current.getString(R.string.redo),
+            if (redoVisibility.value) MaterialTheme.colors.primary else MaterialTheme.colors.secondary
+        ) {
+            if (redoVisibility.value) drawController.reDo()
+        }
+        MenuItems(
+            R.drawable.ic_color,
+            context.getString(R.string.stroke_color),
+            colorValue.value
+        ) {
+            onColorClick()
+        }
+        MenuItems(
+            R.drawable.ic_width,
+            LocalContext.current.getString(R.string.stroke_width),
+            MaterialTheme.colors.primary
+        ) {
+            onWidthClick()
+        }
+        // Erases by drawing over the image in white
+        ToggleButton(
+            R.drawable.black_eraser,
+            R.drawable.white_eraser,
+            context.getString(R.string.toggleEraser),
+            MaterialTheme.colors.primary,
+            onClick = {
+                // Toggle the eraser on or off
+                if (!isToggled.value) {
+                    drawController.changeColor(Color.White)
+                    drawController.changeStrokeWidth(DEFAULT_ERASE_WIDTH)
```

Comment:
Instead of selecting a default erase width, I think it would be better to bind the erase width with the width selector already used for the stroke width.  So I guess simply replacing this line by  `drawController.changeStrokeWidth(currentWidth)` would do the trick.
When using the eraser I was surprised that the erase width was so big compared to the stroke width I was using.

Reply:
Well, to be honest, I highly disagree with you here, but I nevertheless did what you suggested. Your proposal is actually also what I had done initially, but, in my experience, this was not a good idea. The drawing width of the pen always tends to be much smaller than that of the eraser and so when I initially tried out the eraser with the current width, it was a big nuisance seeing that it erased only so little. Then manually having to go and click to first 1) turn on the eraser, 2) click the stroke width changer to adjust the width, 3) change the width by using the slider, 4) erasing, 5) toggling off the eraser to draw again and 6) using the stroke width *again* just to be able to draw with your previous stroke width is not user friendly at all. Usually, when someone needs an eraser in such a small game as ours it’s to make one quick change and they shouldn’t need to worry about it all taking so much time by giving them more options which I personally believe are unnecessary and time-consuming.
So, as I said, I made the requested changes, but I believe this will reduce the user experience and simply be a source of frustration – as it was to me when I tried it initially and which was precisely the reason why I added a default erase width in the first place.

Reply:
Thank you for taking my feedback into account. 
Your arguments make sense and I guess you know better what's best as you tested both implementations. If you want to do it your way I won't refuse, your way of seeing things is totally valid. So I'll approve the PR and it's up to you to decide if you want to use your first implementation before merging.